src/index.js

```diff
@@ -281,13 +281,24 @@ function sanitizeHtml(html, options, _recursing) {
                 parsed = url.parse(value, false, true);
                 var isRelativeUrl = parsed && parsed.host === null && parsed.protocol === null;
                 if (isRelativeUrl) {
-                  // default value of allowIframeRelativeUrls is true unless allowIframeHostnames specified
+                  // default value of allowIframeRelativeUrls is true
+                  // unless allowedIframeHostnames or allowedIframeDomains specified
                   allowed = has(options, "allowIframeRelativeUrls")
-                    ? options.allowIframeRelativeUrls : !options.allowedIframeHostnames;
-                } else if (options.allowedIframeHostnames) {
-                  allowed = options.allowedIframeHostnames.find(function (hostname) {
+                    ? options.allowIframeRelativeUrls
+                    : (!options.allowedIframeHostnames && !options.allowedIframeDomains);
+                } else if (options.allowedIframeHostnames || options.allowedIframeDomains) {
+                  var allowedHostname = (options.allowedIframeHostnames || []).find(function (hostname) {
                     return hostname === parsed.hostname;
                   });
+                  var allowedDomain = (options.allowedIframeDomains || []).find(function(domain) {
+                    // avoid top-level domains
```

Comment:
One last annoying request: please remove the guard. `allowedIframeDomains` is not user input, and I don't see someone putting 'com' in there by accident; we should allow developers to do what they want to do. Consider the case where a custom top level domain is valid only on a certain intranet; here it could make sense.

Reply:
I appreciate the intent though.